### PR TITLE
Jesse merge hosted

### PR DESCRIPTION
**Pull request by jessegoodier**
## What does this PR change?

fixes grafana origin not allowed
simplify hosted cluster deployments
fix agent key templated in values file

## Does this PR rely on any other PRs?

- no

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

Fix for grafana origin not allowed

## Links to Issues or ZD tickets this PR addresses or fixes
NA

## How was this PR tested?
2 test clusters and one beta customer

## Have you made an update to documentation?

NA

### CODE REVIEW

File: cost-analyzer/templates/cost-analyzer-deployment-template.yaml
Comment (michaelmdresser) [inline cost-analyzer/templates/cost-analyzer-deployment-template.yaml:659]: Is it safe to say that `.hosted=true` and `.thanos.query.enabled=true` is a bad config case?

If so, it would be cool if we could somehow enforce this. Some brief Googling found mention of a template function called `fail` [1] but it appears to require some sort of additional library, so I'm not sure if its worth the effort?

[1] https://austindewey.com/2018/12/28/helm-tricks-input-validation-with-required-and-fail/

Comment (mbolt35) [reply cost-analyzer/templates/cost-analyzer-deployment-template.yaml:659]: @michaelmdresser Checkout this PR: https://github.com/kubecost/cost-analyzer-helm-chart/pull/1568

I used `fail` to ensure that a specific combination of flags were set. It works great.

Comment (michaelmdresser) [reply cost-analyzer/templates/cost-analyzer-deployment-template.yaml:659]: Superb. That would be great in this PR if my guess about config state is right.

File: cost-analyzer/values-agent.yaml
Comment (mbolt35) [inline cost-analyzer/values-agent.yaml:13]: Do we not want to set a default name here?

File: cost-analyzer/values-agent.yaml
Comment (mbolt35) [inline cost-analyzer/values-agent.yaml:49]: Same question. Do we not want to set this to a values option?

Comment (jessegoodier) [reply cost-analyzer/values-agent.yaml:49]: can you get this command to work? I can't figure it out. Seems like you can't template in a values file?

```sh
helm template cost-analyzer --repo https://kubecost.github.io/cost-analyzer/ \
 -f https://raw.githubusercontent.com/kubecost/cost-analyzer-helm-chart/develop/cost-analyzer/values-agent.yaml \
 --set agentKeySecretName=myname
```

```
Error: failed to parse https://raw.githubusercontent.com/kubecost/cost-analyzer-helm-chart/develop/cost-analyzer/values-agent.yaml: error converting YAML to JSON: yaml: invalid map key: map[interface {}]interface {}{".Values.agentKeySecretName":interface {}(nil)}
```

Comment (jessegoodier) [reply cost-analyzer/values-agent.yaml:49]: This is the command we share with customers, that is broken with the last agent change PR (which, I think is due to templating in the `values-agent.yaml`):

```
helm install kubecost kubecost/cost-analyzer \
  --namespace kubecost-agent \
--values=https://raw.githubusercontent.com/kubecost/cost-analyzer-helm-chart/develop/cost-analyzer/values-agent.yaml \
  --set prometheus.server.global.external_labels.cluster_id=test \
  --set kubecostProductConfigs.clusterName=test \
  --set-file agentKey="kubecost-agent.key"
```

with my PR, this works again:
```
helm install kubecost kubecost/cost-analyzer \
  --namespace kubecost-agent \
  -f cost-analyzer/values-agent.yaml \
  --set prometheus.server.global.external_labels.cluster_id=test \
  --set kubecostProductConfigs.clusterName=test \
  --set-file agentKey="kubecost-agent.key" \
  --create-namespace
```
With this PR, the secret name could be changed, you would just uncomment that line and edit the line for the thanos-sidecar secret.

Comment (mbolt35) [reply cost-analyzer/values-agent.yaml:49]: I am totally on board with this change, I was just double checking to see if we could somehow require less replication of settings across a values and command line. I still think that's possible, but you guys are working with the customers the most, and if you aren't running into problematic scenarios with the current process, then I fully support it.

Comment (jessegoodier) [reply cost-analyzer/values-agent.yaml:49]: Okay, the PR fixes an issue, and I don't want to wait too long for conflicts. We can address the secret creation when it is required. 

Will merge after approval.

Comment (jessegoodier) [reply cost-analyzer/values-agent.yaml:49]: > Okay, the PR fixes an issue, and I don't want to wait too long for conflicts. We can address the secret creation when it is required.
> 
> Will merge after approval.

Okay, the PR fixes an issue, and I don't want to wait too long for conflicts. We can address the secret creation when it is required.

Will merge after approval.

Comment (jessegoodier) [reply cost-analyzer/values-agent.yaml:49]: > I am totally on board with this change, I was just double checking to see if we could somehow require less replication of settings across a values and command line. I still think that's possible, but you guys are working with the customers the most, and if you aren't running into problematic scenarios with the current process, then I fully support it.

Okay, the PR fixes an issue, and I don't want to wait too long for conflicts. We can address the secret creation when it is required.

Will merge after approval.

File: cost-analyzer/templates/cost-analyzer-deployment-template.yaml
Comment (mbolt35) [inline cost-analyzer/templates/cost-analyzer-deployment-template.yaml:361]: Weird formatting thing happening here?

**Review by mbolt35 — CHANGES_REQUESTED** [review]
Just the weird formatting change here and we'll be good to go: https://github.com/kubecost/cost-analyzer-helm-chart/pull/1620/files#r959970460

File: cost-analyzer/templates/cost-analyzer-deployment-template.yaml
Comment (jessegoodier) [reply cost-analyzer/templates/cost-analyzer-deployment-template.yaml:361]: nice find. not sure what happened there.

**Comment by jessegoodier** [comment]
> Just the weird formatting change here and we'll be good to go: https://github.com/kubecost/cost-analyzer-helm-chart/pull/1620/files#r959970460

fixed.

**Review by mbolt35 — APPROVED** [review]

**Pull request merged**
